router/src/main/java/com/github/zsoltk/compose/savedinstancestate/Persistent.kt

```diff
@@ -1,18 +1,17 @@
 package com.github.zsoltk.compose.savedinstancestate
 
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.MutableState
-import androidx.compose.runtime.state
+import androidx.compose.runtime.remember
 
 @Composable
-fun persistentInt(key: String, defaultValue: Int = 0): MutableState<Int> {
+fun persistentInt(key: String, defaultValue: Int = 0): Int {
     val bundle = AmbientSavedInstanceState.current
 
-    val state: MutableState<Int> = state {
+    val state: Int = remember {
         bundle.getInt(key, defaultValue)
     }
 
-    saveInt(key, state.value)
+    saveInt(key, state)
 
     return state
```

Comment:
You broke the functionality here too. Now these are static and cannot cause a recompose on change anymore.